### ISSUE TITLE
Generates empty `const` statement

### ISSUE THREAD

**Issue by calebeby** [mdx-js/mdx#2112]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

`@mdx-js/mdx@2.1.3`

### Steps to reproduce

My actual use case is that I have an astro site with an mdx file that uses [shiki twoslash](https://shikijs.github.io/twoslash/) to highlight code snippets in the mdx. I haven't been able to narrow it down to a smaller reproducible code example, but I do have a fix!

https://github.com/mdx-js/mdx/blob/main/packages/mdx/lib/plugin/recma-jsx-rewrite.js#L390
This line is generating an empty const statement, which is invalid JS. It should not generate a const statement if there are no declarations.

This fix works:

```js
if (declarations.length > 0) {
  statements.push({
    type: 'VariableDeclaration',
    kind: 'const',
    declarations
  })
}
```

I can send a PR with this change, but I'm not sure how to write a test for this, since I'm not sure how to narrow it down. *Would you be able to help with a test, or would you accept a PR without a test?*

### Expected behavior

Don't emit an empty const declaration
```js
function MDXContent(props = {}) {
  return <MDXLayout {...props}><_createMdxContent {...props} /></MDXLayout>;
}
```

### Actual behavior

It is generating this :(

```js
function MDXContent(props = {}) {
  const ;
  return <MDXLayout {...props}><_createMdxContent {...props} /></MDXLayout>;
}
```

### Runtime

Node v16

### Package manager

npm v8

### OS

macOS

### Build and bundle tools

Astro

**Comment by wooorm** [comment]
It would be *very* nice if there’s a reproduction/test case of this, to prevent it from reoccurring in the future

**Comment by wooorm** [comment]
You might be able to ping the people that work on these tools (astro, orta) to help reduce the test case?

**Comment by fbubbar** [comment]
Coming from [this issue](https://github.com/withastro/compiler/issues/504), I can provide a minimal reproduction of this bug ([Stackblitz](https://stackblitz.com/edit/node-vcfzaq?file=src/pages/index.mdx&view=editor)).

I independently came to the same conclusion as @calebeby and fixed my problem in the same way. If anyone else is experiencing this issue, you can temporarily resolve it by applying this patch:
```patch
diff --git a/lib/plugin/recma-jsx-rewrite.js b/lib/plugin/recma-jsx-rewrite.js
index 1e82a1e41b18b16cb5d2f2a66b07e2d9544c55db..eb5286a273f095bedb693ec587917d6162164108 100644
--- a/lib/plugin/recma-jsx-rewrite.js
+++ b/lib/plugin/recma-jsx-rewrite.js
@@ -387,11 +387,13 @@ export function recmaJsxRewrite(options = {}) {
               })
             }
 
-            statements.push({
-              type: 'VariableDeclaration',
-              kind: 'const',
-              declarations
-            })
+            if (declarations.length > 0) {
+              statements.push({
+                type: 'VariableDeclaration',
+                kind: 'const',
+                declarations
+              })
+            }
           }
 
           /** @type {string} */
```

**Comment by calebeby** [comment]
FWIW I have a PR open that is close... I just started school so I'm super busy right now but I'm planning to get back to it soon

**Comment by wooorm** [comment]
> minimal reproduction of this bug

I’m still looking for something smaller: something without Shiki. The smallest possible reproduction. See the [related PR for more info](https://github.com/mdx-js/mdx/pull/2123#issuecomment-1241087867) :)

**Comment by lloydjatkinson** [comment]
Hello, this might be a good enough repro. It’s using Astro but hopefully that doesn’t really prevent you looking into it. The link to the repro demo is in the linked issue.

https://github.com/withastro/compiler/issues/504#issuecomment-1268959791

**Comment by wooorm** [comment]
I’m looking for a smaller reproduction. The smallest thing possible. So that we can know why it was occurring and that it won’t regress

**Comment by wooorm** [comment]
Released in [`2.1.5`](https://github.com/mdx-js/mdx/releases/tag/2.1.5)!
